### PR TITLE
Add DmqClient ctors accepting Neo.Config arguments

### LINKED ISSUE #8: Add DmqClient ctors accepting Neo.Config arguments
e.g. see https://github.com/sociomantic-tsunami/dhtproto/pull/40

### PR DESCRIPTION
**Pull request by david-eckardt-sociomantic**
Fixes #8

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
I think this requires swarm v4.3.0+ == put it in the neo branch ;)

**Comment by david-eckardt-sociomantic** [comment]
Yeah, I just noticed.

**Comment by david-eckardt-sociomantic** [comment]
Updated: Changed target to the new `neo` branch and updating swarm to v4.4.0.

File: src/dmqproto/client/DmqClient.d [hunk 1]
```diff
@@ -361,6 +396,41 @@ public class SchedulingDmqClient : ExtensibleDmqClient!(RequestScheduler)
...
+                re-established). Of type:
+                void delegate ( IPAddress node_address, Exception e )
+            fiber_stack_size = size (in bytes) of stack of individual connection
+                fibers
+            max_events = limit on the number of events which can be managed
+                by the scheduler at one time. (0 = no limit)
+
+    ***************************************************************************/
+
+    public this ( EpollSelectDispatcher epoll, IClient.Config config,
```
Comment (gavin-norman-sociomantic) [inline src/dmqproto/client/DmqClient.d]: This should accept a `SchedulingDmqClient.Config`. See https://github.com/sociomantic-tsunami/dhtproto/pull/57/files. (Spotted by Joe.)

**Review by gavin-norman-sociomantic — COMMENTED** [review]
LGTM apart from the one comment.

**Comment by david-eckardt-sociomantic** [comment]
Updated.

File: src/dmqproto/client/DmqClient.d [hunk 2]
```diff
@@ -361,6 +396,41 @@ public class SchedulingDmqClient : ExtensibleDmqClient!(RequestScheduler)
...
+
+    /***************************************************************************
+
+        Constructor with support for the neo and legacy protocols. Automatically
+        calls addNodes() with the node definition files specified in the legacy
+        and neo Config instances.
+
+        Params:
+            epoll = EpollSelectDispatcher instance to use
+            config = swarm.client.model.IClient.Config instance. (The Config
```
Comment (gavin-norman-sociomantic) [inline src/dmqproto/client/DmqClient.d]: SchedulingDmqClient.Config

**Comment by david-eckardt-sociomantic** [comment]
Updated. (Somehow Github seems to get the location of the "SchedulingDmqClient.Config" comment https://github.com/sociomantic-tsunami/dmqproto/pull/12#discussion_r148813007 wrong now.)

**Review by gavin-norman-sociomantic — APPROVED** [review]

**Pull request merged**